logstage-elastic/src/main/scala/LogstageCirceElasticRenderingPolicy.scala

```diff
@@ -80,8 +80,17 @@ class LogstageCirceElasticRenderingPolicy(
     result ++= makeEventEnvelope(entry, formatted)
 
     val json = Json.fromFields(result)
+    val maybeTruncatedJson = truncateStringValues.fold(json)(truncateStringValues(json, _))
+
+    dump(maybeTruncatedJson)
+  }
 
-    dump(json)
+  private def truncateStringValues(json: Json, maxLength: Int): Json = {
+    // this is more efficient than json.fold as it avoids unnecessary un- and re-wrapping
+    json
+      .mapObject(_.mapValues(truncateStringValues(_, maxLength)))
+      .mapArray(_.map(truncateStringValues(_, maxLength)))
+      .mapString(string => if (string.length > maxLength) string.take(maxLength) else string)
```

Comment:
should there be an indicator that the text was shrunk? e.g. `[...]`?

Reply:
good point. done